guacamole/src/main/webapp/app/manage/controllers/manageConnectionController.js

```diff
@@ -157,205 +137,178 @@ angular.module('manage').controller('manageConnectionController', ['$scope', '$i
-            !!identifier && (
-               PermissionSet.hasSystemPermission(permissions, PermissionSet.SystemPermissionType.ADMINISTER) || (
-                       PermissionSet.hasConnectionPermission(permissions, PermissionSet.ObjectPermissionType.UPDATE, identifier)
-                   &&  PermissionSet.hasSystemPermission(permissions, PermissionSet.SystemPermissionType.CREATE_CONNECTION)
-               )
-            );
-    
-    }, requestService.WARN);
-   
-    // Get protocol metadata
-    schemaService.getProtocols($scope.selectedDataSource)
-    .then(function protocolsReceived(protocols) {
-        $scope.protocols = protocols;
-    }, requestService.WARN);
-
-    // Get history date format
-    $translate('MANAGE_CONNECTION.FORMAT_HISTORY_START').then(function historyDateFormatReceived(historyDateFormat) {
-        $scope.historyDateFormat = historyDateFormat;
-    }, angular.noop);
-
-    // If we are editing an existing connection, pull its data
-    if (identifier) {
-
-        // Pull data from existing connection
-        connectionService.getConnection($scope.selectedDataSource, identifier)
-        .then(function connectionRetrieved(connection) {
-            $scope.connection = connection;
-        }, requestService.WARN);
+    /**
+     * Loads the data associated with the connection having the given
+     * identifier, preparing the interface for making modifications to that
+     * existing connection.
+     *
+     * @param {String} dataSource
+     *     The unique identifier of the data source containing the connection to
+     *     load.
+     *
+     * @param {String} identifier
+     *     The identifier of the connection to load.
+     *
+     * @returns {Promise}
+     *     A promise which is resolved when the interface has been prepared for
+     *     editing the given connection.
+     */
+    var loadExistingConnection = function loadExistingConnection(dataSource, identifier) {
+        return $q.all({
+            connection     : connectionService.getConnection(dataSource, identifier),
+            historyEntries : connectionService.getConnectionHistory(dataSource, identifier),
+            parameters     : connectionService.getConnectionParameters(dataSource, identifier)
+        })
+        .then(function connectionDataRetrieved(values) {
 
-        // Pull connection history
-        connectionService.getConnectionHistory($scope.selectedDataSource, identifier)
-        .then(function historyReceived(historyEntries) {
+            $scope.connection = values.connection;
+            $scope.parameters = values.parameters;
 
             // Wrap all history entries for sake of display
             $scope.historyEntryWrappers = [];
-            historyEntries.forEach(function wrapHistoryEntry(historyEntry) {
-               $scope.historyEntryWrappers.push(new HistoryEntryWrapper(historyEntry)); 
+            angular.forEach(values.historyEntries, function wrapHistoryEntry(historyEntry) {
```

Comment:
Just curious - why do you prefer `angular.forEach` here instead of the built in array `forEach`?

Reply:
I suppose it's mainly out of habit here. Since `historyEntries` should always be defined, either would work here. Any browsers which would lack `Array.prototype.forEach()` (IE8 and older) are also unable to run Guacamole for more fundamental reasons, and we do use it elsewhere in the webapp.

Were that not the case, `angular.forEach()` has the benefit that it automatically deals with `null` and `undefined` values.

Reply:
I can restore the original usage of built-in `forEach()` if you prefer.

Reply:
I have no problem with `angular.forEach`. :shipit: